src/main/java/org/opensearch/security/ssl/http/netty/Netty4HttpRequestHeaderVerifier.java

```diff
@@ -0,0 +1,135 @@
+import org.opensearch.rest.RestRequest;
+import org.opensearch.security.filter.SecurityRestFilter;
+import org.opensearch.security.http.InterceptingRestChannel;
+import org.opensearch.security.ssl.transport.SSLConfig;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.security.ssl.OpenSearchSecuritySSLPlugin;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.OpenSearchSecurityException;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestResponse;
+
+import java.util.regex.Matcher;
+
+import static org.opensearch.http.netty4.Netty4HttpServerTransport.CONTEXT_TO_RESTORE;
+import static org.opensearch.http.netty4.Netty4HttpServerTransport.EARLY_RESPONSE;
+import static org.opensearch.http.netty4.Netty4HttpServerTransport.SHOULD_DECOMPRESS;
+import static org.opensearch.security.filter.SecurityRestFilter.HEALTH_SUFFIX;
+import static org.opensearch.security.filter.SecurityRestFilter.PATTERN_PATH_PREFIX;
+import static org.opensearch.security.filter.SecurityRestFilter.WHO_AM_I_SUFFIX;
+
+public class Netty4HttpRequestHeaderVerifier extends SimpleChannelInboundHandler<DefaultHttpRequest> {
+    private final SecurityRestFilter restFilter;
+    private final ThreadPool threadPool;
+    private final NamedXContentRegistry xContentRegistry;
+    private final HttpHandlingSettings handlingSettings;
+    private final Settings settings;
+    private final SSLConfig sslConfig;
+    private final boolean injectUserEnabled;
+    private final boolean passthrough;
+
+    public Netty4HttpRequestHeaderVerifier(
+        SecurityRestFilter restFilter,
+        NamedXContentRegistry xContentRegistry,
+        ThreadPool threadPool,
+        HttpHandlingSettings handlingSettings,
+        Settings settings
+    ) {
+        this.restFilter = restFilter;
+        this.xContentRegistry = xContentRegistry;
+        this.threadPool = threadPool;
+        this.handlingSettings = handlingSettings;
+        this.settings = settings;
+
+        this.injectUserEnabled = settings.getAsBoolean(ConfigConstants.SECURITY_UNSUPPORTED_INJECT_USER_ENABLED, false);
+        boolean disabled = settings.getAsBoolean(ConfigConstants.SECURITY_DISABLED, false);
+        if (disabled) {
+            sslConfig = new SSLConfig(false, false);
+        } else {
+            sslConfig = new SSLConfig(settings);
+        }
+        boolean client = !"node".equals(settings.get(OpenSearchSecuritySSLPlugin.CLIENT_TYPE));
+        this.passthrough = client || disabled || sslConfig.isSslOnlyMode();
+    }
+
+    @Override
+    public void channelRead0(ChannelHandlerContext ctx, DefaultHttpRequest msg) throws Exception {
+        // DefaultHttpRequest should always be first and contain headers
+        ReferenceCountUtil.retain(msg);
+
+        if (passthrough) {
+            ctx.fireChannelRead(msg);
+            return;
```

Comment:
does this meant that it'll continue the original behavior: (not handling compressed request efficiently)

Reply:
yes exactly, its as if there is no header verifier step because its just a passthrough. i.e. it gets HttpMessages and fires them to the next handler in the pipeline.